src/index/mod.rs

```diff
@@ -400,10 +418,14 @@ impl IndexBackground {
             return;
         }
         while let Some(index) = self.index.upgrade() {
-            pool.install(|| {
-                optimizing::indexing::optimizing_indexing(index.clone());
-            });
-            std::thread::sleep(Duration::from_secs(60));
+            index.state.store(1, Relaxed);
+            let done = pool.install(|| optimizing::indexing::optimizing_indexing(index.clone()));
+            if done {
```

Comment:
it's possible to continuing creating indexes without new data coming

Reply:
However, the return value of `optimizing_indexing` will only be true if `segs_len < index.options.segment.min_sealed_segment_size as u64 || segs.len() < 3`. This depends on `index.protect`; if `index.protect` is not updated, the condition will always be true. And only `refresh` funtion will update `index.protect`.

Reply:
I'm not sure if `index.state.store(1, Relaxed);` can be reorder after `if`. It's really confusing.

Reply:
update `index.state.store(1, Relaxed);` to `SeqCst` and others to corresponding `Release` and `Acquire`